README.md

```diff
@@ -1,298 +1,433 @@
+Argument(s):
 
-```javascript
-fi = (function() {
-  return {
-    libraryMethod: function() {
-      return "Start by reading the article!";
-    },
-
-    each: function() {
-      /*TODO*/
-    }
-  };
-})();
-
-fi.libraryMethod();
-```
+- a collection (either an object or an array)
+- a callback function
 
-Wrapping a library in code is sometimes called "[The Module Pattern][MP]"
+Return value:
 
-The point of this exercise is to build ***your own implementation*** of the
-collection-processing methods. Don't simply re-use the built-in methods!
-Leverage all you know about callbacks, passing data, etc. to prove that you
-could build your own collection-processing framework whenever ***you*** want.
-
-## Collection Functions (Arrays or Objects)
+- The original collection
 
-**fi.each**
+Behavior:
 
-`fi.each(collection, callback)`
+Iterates over the collection of elements, passing each element in turn to the
+callback function. Returns the original, unmodified, collection for chaining.
 
-Iterates over a **collection** of elements, passing each element in turn to a
-**callback** function. Each invocation of **callback** is called with three
-arguments: (element, index, collection). If **collection** is a JavaScript
-object, **callback**'s arguments will be (value, key, collection). **Returns
-the original collection for chaining.**
+Example function calls:
 
 ```javascript
-fi.each([1, 2, 3], alert);
+myEach([1, 2, 3], alert);
 => alerts each number in turn and returns the original collection
-fi.each({one: 1, two: 2, three: 3}, alert);
+
+myEach({one: 1, two: 2, three: 3}, alert);
 => alerts each number value in turn and returns the original collection
 ```
 
-**fi.map**
+### myMap
+
+`myMap(collection, callback)`
+
+Argument(s):
+
+- a collection (either an object or an array)
+- a callback function
+
+Return value:
+
+- A new array
 
-`fi.map(collection, callback)`
+Behavior:
 
-Produces a new array of values by mapping each value in **collection** through
-a transformation function (**callback**). The callback is passed three
-arguments: the value, then the index (or key) of the iteration, and finally a
-reference to the entire collection. **Returns a new collection for chaining
-without modifying the original.**
+Produces a new array of values by mapping each value in `collection` through a
+transformation function, `callback`. Returns the new array for chaining without
+modifying the original.
+
+Example function calls:
 
 ```javascript
-fi.map([1, 2, 3], function(num){ return num * 3; });
+myMap([1, 2, 3], function(num){ return num * 3; });
 => [3, 6, 9]
-fi.map({one: 1, two: 2, three: 3}, function(num, key){ return num * 3; });
+
+myMap({one: 1, two: 2, three: 3}, function(num, key){ return num * 3; });
 => [3, 6, 9]
 ```
 
-**fi.reduce**
+### myReduce
+
+`myReduce(collection, callback, acc)`
+
+Argument(s):
+
+- a collection (either an object or an array)
+- a callback function
+- a starting value for the accumulator
```

Comment:
Is the accumulator optional? If so, probably helpful to specify that here

Reply:
Done